### PR TITLE
feat(dashboard): Add aggregates and columns fields

### PR DESCRIPTION
**Pull request by shruthilayaj**
Add aggregates and columns fileds to DashboardWidgetQuery
django model to store these values in two separate fields
instead of a single `fields` field.

### CODE REVIEW

File: src/sentry/migrations/0274_add_dashboardwidgetquery_columns_aggregates.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    # we don't want to run in a transaction here, since for long running operations like data
+    # back-fills this results in us locking an increasing number of rows until we finally commit.
+    atomic = False
+
+    dependencies = [
+        ("sentry", "0273_fix_grouplink_seqscans"),
+    ]
+
+    operations = [
+        migrations.AddField(
```
Comment (dashed) [inline src/sentry/migrations/0274_add_dashboardwidgetquery_columns_aggregates.py:32]: You'll want to use `migrations.SeparateDatabaseAndState` here. See https://develop.sentry.dev/database-migrations/#adding-columns

Comment (dashed) [reply src/sentry/migrations/0274_add_dashboardwidgetquery_columns_aggregates.py:32]: I just realized you don't have a default value. So this isn't as dangerous as I had thought.

Comment (shruthilayaj) [reply src/sentry/migrations/0274_add_dashboardwidgetquery_columns_aggregates.py:32]: yup, 👍

File: src/sentry/models/dashboard_widget.py [hunk 2]
```diff
@@ -80,6 +81,12 @@ class DashboardWidgetQuery(Model):
     name = models.CharField(max_length=255)
     fields = ArrayField()
     conditions = models.TextField()
+    # aggregates and columns will eventually replace fields.
+    # Using django's built-in array field here since the one
+    # from sentry/db/model/fields.py adds a default value to the
+    # database migration.
+    aggregates = DjangoArrayField(models.TextField(null=True), null=True)
```
Comment (wedamija) [inline src/sentry/models/dashboard_widget.py:88]: Is `null=True` required for the `TextField`?

**Review by wedamija — APPROVED** [review]
Looks good to me

**Review by wmak — APPROVED** [review]
LGTM

**Comment by shruthilayaj** [comment]
```
--
-- Add field aggregates to dashboardwidgetquery
--
ALTER TABLE "sentry_dashboardwidgetquery" ADD COLUMN "aggregates" text[] NULL;
--
-- Add field columns to dashboardwidgetquery
--
ALTER TABLE "sentry_dashboardwidgetquery" ADD COLUMN "columns" text[] NULL;
```
sql still the same after removing null=True in the base field `TextField`

**Pull request merged**